### PR TITLE
fix(signs): priority of extmark signs

### LINKED ISSUE #19716: Extmarks: signs do not respect `priority`
### Neovim version (nvim -v)

NVIM v0.8.0-dev-813-g6669fc94a

### How to reproduce the issue

Save the following as `test.lua`
```lua
local ns = vim.api.nvim_create_namespace('sign')
vim.api.nvim_buf_set_extmark(0, ns, 0, 0, {sign_text = 'A', priority = 100})
vim.api.nvim_buf_set_extmark(0, ns, 0, 0, {sign_text = 'B', priority = 200})
```
and run `nvim --clean -u test.lua test.lua`.

### Expected behavior

Sign `B` is placed on the first line.

### Actual behavior

Sign `A` is placed on the first line.

### PR DESCRIPTION
**Pull request by lewis6991**
Fixes #19716

### CODE REVIEW

**Comment by lewis6991** [comment]
This also fixes a potential out of bounds array access in the case 9 signs are placed on the same line.

**Comment by clason** [comment]
However, it doesn't fix the situation in #19716 with `signcolumn=auto` (where only a single sign column is available).

**Comment by lewis6991** [comment]
Doesn't it?

**Comment by clason** [comment]
Yep, now it works!

**Comment by clason** [comment]
Can you fix the docs while you're at it (`priority` applies not only to highlights)?

File: test/functional/ui/decorations_spec.lua [hunk 1]
```diff
@@ -1779,6 +1779,25 @@ l5
     ]]}
   end)
 
+  it('works with priority #19716', function()
```
Comment (clason) [inline test/functional/ui/decorations_spec.lua:1782]: Maybe add a second test with `signcolumn=auto` to test correct truncation (as this was a separate fix)?

**Review by clason — APPROVED** [review]

**Pull request merged**